Doc/whatsnew/3.13.rst

```diff
@@ -1787,14 +1787,6 @@ New Deprecations
     Check membership in :data:`~dis.hasarg` instead.
     (Contributed by Irit Katriel in :gh:`109319`.)
 
-* :mod:`getopt` and :mod:`optparse`:
```

Comment:
Do we typically remove from the What's New for the version introduced or do we annotate that it is being changed in 3.14?

Reply:
This is an interesting question. Obviously undeprecation should be retroactive. There is no sense in discouraging the use of getopt and optparse in 3.12 and 3.13. For 3.13, this may even be considered a bugfix.

But how to indicate this in What's New? If say nothing, then users will continue to think that getopt and optparse are deprecated. If write about this in 3.12 and 3.13, most users will not read this. So we should write something in 3.14, even if this change will be applied to 3.12 and 3.13.

Reply:
I don't have a good answer to this. Perhaps @hugovk would like to weigh in as release manager.

Reply:
@Yhg1s is the 3.13 RM so punting to him :) 

However, optparse was deprecated back in 3.2, some 13 years ago, so for one of them, this isn't a sudden change in 3.13.

If we do undeprecate, perhaps adding a note to the existing "What's New in 3.13" entry to note it's future change?

In any case, I would also hold off merging this and allow some more time for the discussion, it's not yet been 24 hours:

https://discuss.python.org/t/getopt-and-optparse-vs-argparse/69618

Reply:
In drafting my PR, I took the view that 3.13 is new enough that most of the people who *will* read the What's New haven't actually read it yet, and hence just changed this to a regular 3.13 What's New entry for `optparse` (`getopt` is reverting to the 3.12 status quo, so it only got mentioned in NEWS).